vcluster/configure/vcluster-yaml/control-plane/components/coredns.mdx

```diff
@@ -9,43 +9,176 @@ description: Configure CoreDNS.
+
+When a pod queries a service such as `my-app.default.svc.cluster.local`, CoreDNS returns the service IP address for normal service discovery.
+
+## How DNS resolution works in virtual clusters
+
+DNS resolution in virtual clusters works the same as standard Kubernetes clusters. Applications make DNS queries to resolve service names, and CoreDNS responds with the correct IP addresses for services within the virtual cluster.
+
+When an application pod initiates a DNS query, the request travels to the virtual cluster's CoreDNS instance on port 1053. CoreDNS maintains a complete view of all services that exist within the virtual cluster and resolves queries to the appropriate service IP addresses.
+
+## Deployment architectures
+
+Virtual clusters support two CoreDNS deployment options.
+
+### Deploy separate CoreDNS
+
+A standard virtual cluster deployment consists of two pods per virtual cluster instance: the virtual cluster pod containing the API server and syncer components, and a separate CoreDNS pod. Each virtual cluster operates its own DNS service so that pods and services can locate each other by hostnames, with the syncer ensuring that standard Kubernetes DNS naming logic applies.
+
+The separate deployment model isolates DNS operations from the API server and sync functions. CoreDNS listens on port 1053 instead of port 53 to avoid conflicts with the host cluster's DNS service. DNS issues can be debugged, updated, or restarted independently without affecting the virtual cluster's control plane availability. However, virtual cluster pods cannot directly access host cluster services via DNS due to isolation boundaries.
+
+### Deploy embedded CoreDNS
+
 <ProAdmonition/>
 
-## Separate CoreDNS
+The embedded CoreDNS feature allows you to run CoreDNS as part of the syncer, which saves the overhead of an external CoreDNS pod. This enterprise feature consolidates all virtual cluster components into a single pod, reducing pod count and optimizing resource allocation.
+
+The embedded approach requires consideration of operational trade-offs. Troubleshooting might become more complex since DNS and sync functions share the same container and process space. Updates and restarts affect both DNS and sync capabilities simultaneously, though all CoreDNS features remain supported.
+
+## CoreDNS uses port 1053
+
+CoreDNS in virtual clusters listens on port 1053 instead of the default DNS port 53 to avoid conflicts with the host cluster's DNS.
+
+The host cluster already operates its own DNS service on the standard port 53, so the virtual cluster DNS services cannot bind to the same port. Since multiple virtual clusters might run on the same host cluster, each requires its own DNS service that operates independently.
+
+Port 53 is classified as a privileged port on most operating systems and requires elevated permissions to bind to it. By using port 1053, CoreDNS containers can operate with standard user privileges.
```

Comment:
This - `to avoid conflicts with the host cluster's DNS`, and the second paragraph, is not accurate. Each pod has it's own network namespace, so there are no port conflicts between apps deployed in k8s. 
The actual reason is what is explained in the 3rd paragraph.